Code/src/megadesk.cpp

```diff
@@ -747,162 +757,135 @@ void loop()
+  }
+  else if (targetHeight < currentHeight - HYSTERESIS && currentHeight > minHeight) {
+#if (defined SERIALCOMMS && defined DEBUGHEIGHT)
+    writeSerial(response_down, targetHeight);
+#endif
     MOTOR_DOWN;
+  }
   else
   {
     // some possibilities:
-    //   close enough and still coasting,
+    //   close enough (maybe still coasting),
     //   or overshot,
     //   or out of range!
     // so stop
     MOTOR_OFF;
-    memoryMoving = false;
   }
 
 }
 
+// check if motor is idle
+bool isIdle(byte value) {
+  return (value == LIN_MOTOR_IDLE1 || value == LIN_MOTOR_IDLE2 || value == LIN_MOTOR_IDLE3);
+}
+
 uint8_t linBurst()
 {
   static byte empty[3] = {0, 0, 0};
-  static byte cmd[3] = {0, 0, 0};
-  static byte node_a[3] = {0, 0, 0};
-  static byte node_b[3] = {0, 0, 0};
+  static byte cmd[3];
+  byte node_a[3];
+  byte node_b[3];
   static State lastState = State::OFF;
+  byte lin_cmd = LIN_CMD_IDLE;
 
   // Send PID 17
   lin.send(17, empty, 3);
 
   delayUntil(5);
   // Recv from PID 09
   uint8_t chars = lin.recv(9, node_b, sizeof(node_b));
+  uint16_t enc_b = node_b[0] | (node_b[1] << 8);
   if (chars != sizeof(node_b)+1) {
 #ifdef SERIALERRORS
 #ifdef SERIALCOMMS
-    writeSerial(response_error, node_b[0] | (node_b[1] << 8), chars, badPID9Marker);
+    writeSerial(response_error, enc_b, chars, badPID9Marker);
 #endif
 #ifdef FEEDBACK
     if (feedback) playTone(NOTE_A4, PIP_DURATION);
 #endif
 #endif
     return 0;
   }
+
   delayUntil(5);
   // Recv from PID 08
   chars = lin.recv(8, node_a, sizeof(node_a));
+  uint16_t enc_a = node_a[0] | (node_a[1] << 8);
   if (chars != sizeof(node_a)+1) {
 #ifdef SERIALERRORS
 #ifdef SERIALCOMMS
-    writeSerial(response_error, node_a[0] | (node_a[1] << 8), chars, badPID8Marker);
+    writeSerial(response_error, enc_a, chars, badPID8Marker);
 #endif
 #ifdef FEEDBACK
     if (feedback) playTone(NOTE_A4, PIP_DURATION);
 #endif
 #endif
     return 0;
   }
+
   // Send PID 16, 6 times
   for (byte i = 0; i < 6; i++)
   {
     delayUntil(5);
     lin.send(16, 0, 0);
   }
 
-  delayUntil(5);
-  // Send PID 1
-  lin.send(1, 0, 0);
+  uint16_t enc_target = enc_a; //(enc_a+enc_b)/2;
+  uint16_t enc_min = min(enc_a, enc_b);
+  uint16_t enc_max = max(enc_a, enc_b);
+  currentHeight = enc_target;
 
-  uint16_t enc_a = node_a[0] | (node_a[1] << 8);
-  uint16_t enc_b = node_b[0] | (node_b[1] << 8);
-  uint16_t enc_target = enc_a;
-  currentHeight = enc_a;
-#if (defined SERIALCOMMS && defined SERIALERRORS)
-  // monitor delta between motor positions. report if above 20.
-  if ( min(enc_a-enc_b, enc_b-enc_a) > 20)
-    writeSerial(response_error, enc_a, min(enc_a-enc_b, enc_b-enc_a), DriftMarker);
+  // monitor drift between motor positions. report if above 20.
+  if ( enc_max - enc_min > 20) {
+#ifdef SERIALERRORS
+#ifdef SERIALCOMMS
+    writeSerial(response_error, enc_min, enc_max-enc_min, DriftMarker);
 #endif
-
-  // Send PID 18
-  switch (state)
```

Comment:
all logic in this switch block is moved into the second switch. 
the lin.send(18,cmd,3) is the last lin communication and now happens at the very end of this function.